inception-ui-external-search/src/main/java/de/tudarmstadt/ukp/inception/app/ui/externalsearch/sidebar/ExternalSearchAnnotationSidebar.java

```diff
@@ -0,0 +1,427 @@
+        mainContainer.setOutputMarkupId(true);
+        add(mainContainer);
+
+        DocumentRepositorySelectionForm projectSelectionForm = new DocumentRepositorySelectionForm(
+            "repositorySelectionForm");
+        mainContainer.add(projectSelectionForm);
+
+        SearchForm searchForm = new SearchForm("searchForm");
+        add(searchForm);
+        mainContainer.add(searchForm);
+
+        List<IColumn<ExternalSearchResult, String>> columns = new ArrayList<>();
+
+        columns.add(new AbstractColumn<ExternalSearchResult, String>(new Model<>("Results"))
+        {
+            @Override public void populateItem(Item<ICellPopulator<ExternalSearchResult>> cellItem,
+                String componentId, IModel<ExternalSearchResult> model)
+            {
+                @SuppressWarnings("rawtypes") Item rowItem = cellItem.findParent(Item.class);
+                int rowIndex = rowItem.getIndex();
+                ResultRowView rowView = new ResultRowView(componentId, rowIndex + 1, model);
+                cellItem.add(rowView);
+            }
+        });
+
+        if (searchState.getDataProvider() == null) {
+            searchState.setDataProvider(new ExternalResultDataProvider(externalSearchService,
+                userRepository.getCurrentUser(),
+                searchStateModel.getObject().getCurrentRepository(), ""));
+        }
+
+        dataTableContainer = new WebMarkupContainer("dataTableContainer");
+        dataTableContainer.setOutputMarkupId(true);
+        mainContainer.add(dataTableContainer);
+
+        DataTable<ExternalSearchResult, String> resultTable = new DefaultDataTable<>("resultsTable",
+            columns, searchState.getDataProvider(), 8);
+        resultTable.setCurrentPage(searchState.getCurrentPage());
+        dataTableContainer.add(resultTable);
+
+    }
+
+    @Override protected void onDetach()
+    {
+        // Save the current page number of the search results when the sidebar being switched
+        DataTable<ExternalSearchResult, String> resultTable =
+            (DataTable<ExternalSearchResult, String>) dataTableContainer.get("resultsTable");
+        searchStateModel.getObject().setCurrentPage(resultTable.getCurrentPage());
+        super.onDetach();
+    }
+
+    @OnEvent public void onRenderAnnotations(RenderAnnotationsEvent aEvent)
+    {
+        if (selectedResult != null) {
+            // TODO highlight keyword
+        }
+    }
+
+    private class DocumentRepositorySelectionForm
+        extends Form<DocumentRepository>
+    {
+        public DocumentRepositorySelectionForm(String aId)
+        {
+            super(aId);
+
+            DropDownChoice<DocumentRepository> repositoryCombo =
+                new BootstrapSelect<DocumentRepository>("repositoryCombo",
+                new PropertyModel<DocumentRepository>(ExternalSearchAnnotationSidebar.this,
+                    "searchStateModel.getObject().getCurrentRepository()"), repositoriesModel)
```

Comment:
> "searchStateModel.getObject().getCurrentRepository()"

Does this really work? I would rather have expected something like ""searchStateModel.currentRepository". Instead of fetching the model from the outer class, I would propose to pass it as a constructor argument. Optimally, it should be possible that the form class is static.  In fact, in other parts of the code, I am moving away from form classes and instead construct the entire form inline or in a helper method.

Reply:
I don't understand what you are saying here, so let me see if this is what you are looking for: I followed the same way `SearchSidebar` saves its state, where `searchStateModel` is type `CompoundPropertyModel<ExternalSearchUserState>`. So I need to call `getObject()` get get the `ExternalSearchUserState`.

Reply:
Yes, but here you are trying to put Java code into a String which isn't intended to do Java code. The PropertyModel operates on *[Property Expressions](https://cwiki.apache.org/confluence/display/WICKET/Property+Expression+Language)*. If you already have a CompoundPropertyModel, you can also use the [`bind`](https://cwiki.apache.org/confluence/display/WICKET/Working+with+Wicket+models#WorkingwithWicketmodels-CompoundPropertyModels) method which doesn't require you to create yet another model (search for ` CompoundPropertyModel.bind`). `bind` also uses *Property Expressions*.

Reply:
Oh I see what you mean. It must have happened when I did a replaceAll.